comp/remote-config/rcclient/rcclientimpl/rcclient.go

```diff
@@ -83,8 +83,11 @@ func newRemoteConfigClient(deps dependencies) (rcclient.Component, error) {
 		client.WithPollInterval(5 * time.Second),
 	}
 
-	if deps.Params != nil {
-		optsWithDefault = append(optsWithDefault, deps.Params)
+	if deps.Params.AgentName != "" && deps.Params.AgentVersion != "" {
+		optsWithDefault = append(
+			optsWithDefault,
+			client.WithAgent(deps.Params.AgentName, deps.Params.AgentVersion),
+		)
```

Comment:
The condition here only append the `client.WithAgent` if both agent name and agent version are not empty. Is that on purpose? 

What if we modify the code to return an error in the case that any of the params are empty? We do not want a RC client of `"unknown"` type, right? 

What about this?

```go
if deps.Params.AgentName == "" || deps.Params.AgentVersion = "" {
	return nil, fmt.Error("RC params agent name or agent verion are empty. Ensure both are provided")
}

// Append client options
optsWithDefault := []func(*client.Options){
	client.WithPollInterval(5 * time.Second),
	client.WithAgent(deps.Params.AgentName, deps.Params.AgentVersion)
}
```

The error message is not ideal 😉